docs/sphinx/releases/whats_new_1_4_0.rst

```diff
@@ -14,11 +14,59 @@
 General changes
 ===============
 
+* We have added the collectives ``all_to_all`` and ``all_reduce``.
+* Components can now be checkpointed.
+* We have added the performance counters ``background-receive-duration`` and
+  ``background-receive-overhead`` for inspecting the time and overhead spent on
+  receiving parcels in the backgroun d.
+* Compilation time has been further improved when ``HPX_HAVE_NETWORKING=OFF``.
+* We no longer require compiled Boost dependencies in certain configurations.
+  This requires at least Boost 1.70, compiling on x86 with GCC 9 or newer in
+  C++17 mode. The dependency on Boost.Filesystem can explicitly be turned on
+  with ``HPX_FILESYSTEM_WITH_BOOST_FILESYSTEM_COMPATIBILITY=ON`` (it is off by
+  default if the standard library supports ``std::filesystem``).
+  Boost.ProgramOptions has been copied into the HPX repository. We have a
+  compatibility layer for users who must explicitly use Boost.ProgramOptions
+  instead of the ProgramOptions provided by HPX. To remove the dependency
+  ``HPX_PROGRAM_OPTIONS_WITH_BOOST_PROGRAM_OPTIONS_COMPATIBILITY`` must be
+  explicitly set to ``OFF``. This option will be removed in a future release.
+* It is now possible to use the process affinity mask set by tools like
+  ``numactl`` and various batch environments with the command line option
+  :option:`--hpx:use-process-mask`.
+* It is now possible to create standalone thread pools without starting the
+  runtime. See the ``standalone_thread_pool_executor.cpp`` test in the
+  ``execution`` module for an example.
+* Cloning of APEX was defective in previous releases (it required manual
+  intervention to check out the correct tag or branch). This has been fixed.
+* The option ``HPX_WITH_MORE_THAN_64_THREADS`` is now ignored and will be
+  removed in a future release. The value is instead derived directly from
+  ``HPX_WITH_MAX_CPU_COUNT`` option.
+* We have deprecated compiling in C++11 mode. The next release will require a
+  C++14 capable compiler.
+* We have deprecated support for the Vc library. This option will be replaced
+  with SIMD support from the standard library in a future release.
+* We have significantly refactored our CMake setup. This is intended to be a
+  non-breaking change and will allow for using HPX through CMake targets in the
+  future.
+* We have continued modularizing the HPX library. In the process we have
+  rearranged many header files into module-specific directories. All moved
+  headers have compatibility headers which forward from the old location to the
+  new location, together with a deprecation warning. The compatibility headers
+  will eventually be removed.
+
```

Comment:
Here is a list of more changes (in no particular order): stack-less coroutines, unique_any and friends, SPDX licensing tags, reworking cmake (targets!), clang-formatting modules, resiliency APIs, performance improvements (mostly scheduler, etc.), there might be more...

Also, what compilers are deprecated with this release? gcc v5, clang v???

Reply:
I've missed so many things (big release!). Will definitely add the stackless coroutines, unique_any etc., resiliency, the performance improvements, and SPDX license tag. I've mentioned the CMake targets but don't want to advertize it too much before #3818. Is clang-formatting relevant to users?

Reply:
> Is clang-formatting relevant to users?

Probably not so much if taken in isolation. It's more of a QOI issue that underlines our efforts towards a homogeneous user experience.

Reply:
Fair point. I'll add it. Thanks!

Reply:
@msimberg Should we add a (small) paragraph elaborating on the modularization efforts? I mean talking about non-cyclicity, maintainability, the possibility to use separate modules without pulling in all of HPX (serialization!), etc.. We might want to outline the overall goal of this refactoring, the possible (and already real) benefits of going through this. The modularization is a major part of this release, it will also be _the_ motivation for releasing HPX V2.0 (once done).

Reply:
Yes, but no!? It's important but I don't think it belongs in the release notes in that detail. There is a note on modularization and the practical implications of headers moving around. Again I don't want to advertise the targets too much before we finalize the names, namespacing, static/shared libraries, granularity etc. unless we make it clear that it's "experimental" in the sense that things might still move around. We should definitely advertise the whole effort as much as possible though in the release announcement and everywhere else! It's probably even worthy of a blog post on its own. What do you think?

Reply:
> It's probably even worthy of a blog post on its own. 

Absolutely! Otherwise, I'm fine with not expanding it here too much.